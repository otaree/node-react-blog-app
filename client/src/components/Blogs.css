.Blogs.__container {
    width: 80%;
    margin: 10px auto;
}

.Blogs.__container ul {
    list-style: none;
}

.Blogs.__container ul a {
    text-decoration: none;
    color: #222;
    font-size: 18px;
    font-weight: bold;
}

.Blogs.__container ul a:hover {
    color: #ddd;
}