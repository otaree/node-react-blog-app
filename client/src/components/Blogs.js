import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import './Blogs.css';

export const Blogs = (props) => (
    <div className="Blogs __container">
        {
            props.blogs.length > 0 ?
            (
            <ul>
                {
                    props.blogs.map(blog => <li key={blog.id}><Link  to={`/blogs/${blog.id}`}>{blog.title}</Link></li>)
                }
            </ul>
            ) :
            <h3 style={{ textAlign: 'center'}}>No Blog found</h3> 
        }
    </div>
);


const mapStateToProps = state => {
    return {
        blogs: state.blogs.map(blog => {
            return {
                id: blog._id,
                title: blog.title
            }
        })
    };
}

export default connect(mapStateToProps)(Blogs);