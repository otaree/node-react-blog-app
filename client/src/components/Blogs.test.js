import React from 'react';
import { shallow } from 'enzyme';
import { Link } from 'react-router-dom';
import { Blogs } from './Blogs';

describe('Blogs', () => {
    const props = {
        blogs: [{
        id: '1',
        title: "First title"
    }, {
        id: '2',
        title: "Second title"
    }]
    };

    const blogs = shallow(<Blogs  {...props} />);

    it('render properly', () => {
        expect(blogs).toMatchSnapshot();
    });

    describe('when there are valid blogs props', () => {
        it('should show correct numbers of `Link`', () => {
            expect(blogs.find(Link).length).toEqual(2);
        });
    });
});


