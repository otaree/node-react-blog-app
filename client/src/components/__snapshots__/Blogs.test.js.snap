// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blogs render properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    blogs={
        Array [
            Object {
              "id": "1",
              "title": "First title",
            },
            Object {
              "id": "2",
              "title": "Second title",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <ul>
        <li>
                <Link
                        replace={false}
                        to="/blogs/1"
                >
                        First title
                </Link>
        </li>
        <li>
                <Link
                        replace={false}
                        to="/blogs/2"
                >
                        Second title
                </Link>
        </li>
</ul>,
      "className": "Blogs __container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            <Link
                        replace={false}
                        to="/blogs/1"
            >
                        First title
            </Link>
</li>,
          <li>
            <Link
                        replace={false}
                        to="/blogs/2"
            >
                        Second title
            </Link>
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/blogs/1"
>
              First title
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "First title",
              "replace": false,
              "to": "/blogs/1",
            },
            "ref": null,
            "rendered": "First title",
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/blogs/2"
>
              Second title
</Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "Second title",
              "replace": false,
              "to": "/blogs/2",
            },
            "ref": null,
            "rendered": "Second title",
            "type": [Function],
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <ul>
          <li>
                    <Link
                              replace={false}
                              to="/blogs/1"
                    >
                              First title
                    </Link>
          </li>
          <li>
                    <Link
                              replace={false}
                              to="/blogs/2"
                    >
                              Second title
                    </Link>
          </li>
</ul>,
        "className": "Blogs __container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li>
              <Link
                            replace={false}
                            to="/blogs/1"
              >
                            First title
              </Link>
</li>,
            <li>
              <Link
                            replace={false}
                            to="/blogs/2"
              >
                            Second title
              </Link>
</li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/blogs/1"
>
                First title
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "First title",
                "replace": false,
                "to": "/blogs/1",
              },
              "ref": null,
              "rendered": "First title",
              "type": [Function],
            },
            "type": "li",
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "host",
            "props": Object {
              "children": <Link
                replace={false}
                to="/blogs/2"
>
                Second title
</Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "Second title",
                "replace": false,
                "to": "/blogs/2",
              },
              "ref": null,
              "rendered": "Second title",
              "type": [Function],
            },
            "type": "li",
          },
        ],
        "type": "ul",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
